index/models.py

```diff
@@ -119,51 +122,64 @@ def add_info_other(self, info):
 class Petition(models.Model):
     title = models.CharField(max_length=200)  # Title of petition
     description = models.CharField(max_length=4000)  # The paragraph description
-    ID = models.IntegerField(999999, primary_key=True)
+    ID = models.IntegerField(primary_key=True)
     archived = models.BooleanField(default=False)
     hidden = models.BooleanField(default=False)
 
-    # Files the date created
-    created_date = models.DateTimeField(db_index=True, default=timezone.now())
-    # The expected signature to move to the next step
-    expected_sig = models.IntegerField(300)
-    # The author of the petition
-    author = models.ForeignKey(User, on_delete=models.CASCADE, related_name="petitions")
-    # The tags, probably a max of 3
-    tags = models.ManyToManyField(Tag, related_name="petitions")
-    # The signature
-    signatures = models.ManyToManyField(Signature, related_name="petitions")
-    # If the senate has responded , their answer
+    created_date = models.DateTimeField(
+        db_index=True, default=timezone.now()
+    )  # Files the date created
+    expected_sig = models.IntegerField(
+        300
+    )  # The expected signature to move to the next step
+
+    author = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="petitions", blank=True, null=True
+    )  # The author of the petition
+    tags = models.ManyToManyField(
+        Tag, related_name="petitions", blank=True, null=True
+    )  # The tags, probably a max of 3
+    signatures = models.ManyToManyField(
+        Signature, related_name="petitions", blank=True
+    )  # The signature
+
     senate_response = models.ForeignKey(
-        Response, on_delete=models.CASCADE, related_name="petitions"
-    )
+        Response,
+        on_delete=models.CASCADE,
+        related_name="petitions",
+        blank=True,
+        null=True,
+    )  # If the senate has responded , their answer
 
     # Returns title when asked for the item
     def __unicode__(self):
         return self.title
 
-    # Sets the hidden variable as true if we don't want it to
-    # Be displayed in the main site
+        # Sets the hidden variable as true if we don't want it to
+        # Be displayed in the main site
+
     def set_hidden(self, bool_val):
         self.hidden = bool_val
```

Comment:
These setters are unnecessary imo. They don't add anything above just assigning to the fields from outside the object.

Reply:
So in that case I'd prefer using a `@property` decorator so that setters and getters are triggered on property access: https://docs.python.org/3/howto/descriptor.html#properties